packages/http/package.json

```diff
@@ -30,6 +30,7 @@
     "faker": "^5.5.3",
     "fp-ts": "^2.11.5",
     "json-schema-faker": "0.5.0-rcv.40",
+    "json-schema-merge-allof": "^0.8.1",
```

Comment:
Although Stoplight has a fork of `json-schema-merge-allof` it introduces some new behavior ([discarding `$ref`s](https://github.com/stoplightio/json-schema-merge-allof/blob/85cc8c7ee7d944aa117f173c11b0bc1087c36dae/src/index.js#L609-L611)) that breaks an existing test.  I believe that is a "true" failure that reveals an undesirable change.  This prompts a couple questions:

1. Is it necessary that we use the Stoplight branch?
2. If so, is it possible to use the Stoplight branch without the changes?
3. If the custom behaviors can't be disabled, do I need to resolve references (via `json-schema-ref-parser` or similar) prior to using Stoplight's merge-allOf?

Reply:
I'm not sure that I'm the right person to answer this question, but I'll try:

1. We use `json-schema-merge-allof` as well in `public-api` and `studio` packages. Both are using our fork.
2. I'd say we should use the same version everywhere for consistency.
3. This is a good question. The schema should be dereferenced by the `getHttpOperationsFromSpec` method, I'm not sure this is required in this particular case.

Reply:
@P0lip, after talking with @rainum today, my tentative plan is to rely on Stoplight's fork of the _json-schema-merge-allof_ library here instead of the "original".  However, the recently added behavior in our fork that [discards $refs](https://github.com/stoplightio/json-schema-merge-allof/blob/85cc8c7ee7d944aa117f173c11b0bc1087c36dae/src/index.js#L609-L611) is not desirable in this situation, so I think I'll need to add some kind of switch/configuration to our fork.

How would you approach this?  Is what I propose a reasonable solution?

Reply:
You can provide a custom $ref resolver, i.e. https://github.com/stoplightio/json-schema-tree/blob/master/src/mergers/mergeAllOf.ts

Reply:
Thank you, @P0lip !  That's perfect.